### ISSUE TITLE
envoy send RST tcp segment

### ISSUE THREAD

**Issue by liuqi-sun** [envoyproxy/envoy#15868]
After three handshakes, the HTTP request is sent, and the RST fragment is returned

![envoy-rst](https://user-images.githubusercontent.com/48784730/113853123-e1f6bb80-97cf-11eb-8ef5-e3fed5b5f7d1.png)

**Comment by snowp** [comment]
Sorry can you elaborate on this some more? Configuration used, how you are reproducing this, what you expect, etc.

**Comment by liuqi-sun** [comment]
> Sorry can you elaborate on this some more? Configuration used, how you are reproducing this, what you expect, etc.

Envoy acts as the data control surface for Istio,Istio redirects the port through the iptable to 15006. The envoy listens to 15006. But when the client connects the envoy through 15006, the result envoy sends RST.
![envoy-rst-1](https://user-images.githubusercontent.com/48784730/113962882-2aa78680-985b-11eb-9f84-1c9d1f8d9188.png)
![envoy-rst-2](https://user-images.githubusercontent.com/48784730/113962884-2c714a00-985b-11eb-8d1c-2c3673247f03.png)

**Comment by snowp** [comment]
Sorry there's not much I can do without more information about the config Envoy is using. I'm guessing there is a misconfiguration that causes Envoy to reset the incoming request. 

I suggest bringing this to the Istio folks brings to help understand what configuration Envoy is using here.

**Comment by lambdai** [comment]
The tcp hand shake is complete: iptables says yes. It's very likely the envoy listener is ready.

Could you turn on the access log at envoy? With the access log you can understand if a filter chain is found or not. If the filter chain is selected, usually the detailed error flag is revealed in the access log

**Comment by liuqi-sun** [comment]
> The tcp hand shake is complete: iptables says yes. It's very likely the envoy listener is ready.
> 
> Could you turn on the access log at envoy? With the access log you can understand if a filter chain is found or not. If the filter chain is selected, usually the detailed error flag is revealed in the access log

thank you

**Comment by liuqi-sun** [comment]
> Sorry there's not much I can do without more information about the config Envoy is using. I'm guessing there is a misconfiguration that causes Envoy to reset the incoming request.
> 
> I suggest bringing this to the Istio folks brings to help understand what configuration Envoy is using here.

thank you

**Comment by liuqi-sun** [comment]
> Sorry there's not much I can do without more information about the config Envoy is using. I'm guessing there is a misconfiguration that causes Envoy to reset the incoming request.
> 
> I suggest bringing this to the Istio folks brings to help understand what configuration Envoy is using here.

When you access an envoy using iptables to redirect ports, does an envoy have the ability to disable requests from an upstream service that does not exist on a port?For example, the upstream service listens on 80, which I can only access through 80, even though the port is redirected

**Comment by liuqi-sun** [comment]
> The tcp hand shake is complete: iptables says yes. It's very likely the envoy listener is ready.
> 
> Could you turn on the access log at envoy? With the access log you can understand if a filter chain is found or not. If the filter chain is selected, usually the detailed error flag is revealed in the access log

When you access an envoy using iptables to redirect ports, does an envoy have the ability to disable requests from an upstream service that does not exist on a port?For example, the upstream service listens on 80, which I can only access through 80, even though the port is redirected

**Comment by lambdai** [comment]
> When you access an envoy using iptables to redirect ports, does an envoy have the ability to disable requests from an upstream service that does not exist on a port?For example, the upstream service listens on 80, which I can only access through 80, even though the port is redirected

Sorry I cannot understand. Can you rephrase?
Meanwhile please check if bind_to_port works for you https://github.com/envoyproxy/envoy/blob/58606021040b5719db9b6ff3d18ceebdff50fcda/api/envoy/config/listener/v3/listener.proto#L276
